### ISSUE TITLE
HDR mode error with the latest Nvidia driver (45x.xx)

### ISSUE THREAD

**Issue by hooke007** [mpv-player/mpv#7866]
### Messages
- mpv version
mpv-x86_64-20200621-git-dc24a43
- Windows Version
win10 x64 1909
- Others
My monitor (supports HDR1000) will automatically transferred into the HDR mode by the following conf.
After I updated the latest NVidia driver (451.48 WHQL), it can't enter the HDR mode any more.

### Reproduction steps
Here is my conf:
> vo=gpu
> gpu-api=d3d11
> hwdec=no
> profile=gpu-hq
> fs
> dither-depth=10
> d3d11-output-format=rgb10_a2
> d3d11-output-csp=pq
> d3d11-exclusive-fs=yes
> interpolation
> video-sync=display-resample
> tscale=oversample

### Expected behavior
In HDR mode, the local dimming is activated.
### Actual behavior
Incorrect color and no local dimming.
### Log file
[https://github.com/hooke007/temp/blob/master/mpv-20200630-1407.log](url)

**Comment by CounterPillow** [comment]
>  After I updated the latest NVidia driver (451.48 WHQL), it can't enter the HDR mode any more.

Sounds like something you should report to nvidia if this changed with an update of the nvidia driver.

**Comment by hooke007** [comment]
> Sounds like something you should report to nvidia if this changed with an update of the nvidia driver.

I found my madvr didn't work neither.
http://bugs.madshi.net/view.php?id=647
If it is true, maybe mpv also need to make a change?

**Comment by jeeb** [comment]
Yes, the driver changing whatever it's doing most likely is going to affect applications which utilize this stuff. Thus madVR is also affected. That's all. The reporter of the issue for madVR most likely has no idea about things, either.

What an application can do is set its swap chain color space, _which according to the log is still correct and succeeds_:
```
[   0.080][v][vo/gpu/d3d11] Selected swapchain format R10G10B10A2_UNORM (24), attempting to utilize it.
[   0.081][v][vo/gpu/d3d11] Selected swapchain color space RGB_FULL_G2084_NONE_P2020 (12), attempting to utilize it.
[   0.081][v][vo/gpu/d3d11] Swapchain capabilities for color space RGB_FULL_G2084_NONE_P2020 (12): normal: yes, overlay: no
[   0.081][v][vo/gpu/d3d11] Swapchain successfully configured to color space RGB_FULL_G2084_NONE_P2020 (12)!
```

For me the behavior has always been:
* In windowed mode the desktop stays however it is (SDR or HDR).
* In full-screen (even w/o exclusive FS) Windows **or** the driver attempts to switch the color space (even if the monitor cannot do it - which can lead to hilarious things).

This is at least until 442.92.

Anyways, please test with just `--no-config --d3d11-output-csp=pq --log-file=mpv_sucks.log`. And unfortunately unless nvidia's change log mentions anything specific, this is an nvidia bug since as far as I can tell the software is just setting the swap chain color space and that succeeds.

**Comment by hooke007** [comment]
> Anyways, please test with just `--no-config --d3d11-output-csp=pq --log-file=mpv_sucks.log` 

https://github.com/hooke007/temp/blob/master/mpv_sucks.log
I have tested again and it didn't work.

**Comment by Doofussy2** [comment]
There're likely some changes due to DirectX Ultimate. That's the first driver that supports it.

**Comment by hooke007** [comment]
> > Sounds like something you should report to nvidia if this changed with an update of the nvidia driver.
> 
> I found my madvr didn't work neither.
> http://bugs.madshi.net/view.php?id=647
> 
> > since nvidia changed the hdr code and/or hdr handling in 45x.xx driver branch madvr hdr passthrough does not work anymore.
> > instead of passthroughing hdr to the display madvr activates windows hdr and sends sdr to display.
> 
> If it is true, maybe mpv also need to make a change?

It seems Nvidia did sth terrible , which will result in the HDR output not being lossless.
I have to return back to the old driver (442.xx)

**Comment by hooke007** [comment]
Mark. 456.38 doesn't work yet. (Though it have fixed the issue about Madvr's HDR passthrough.)

**Comment by ValZapod** [comment]
Actually it does passthrough metadata when HDR is preactivated.

**Comment by kasper93** [comment]
I'm closing this, because we are not NVIDIA support.

Let me know if there are more issues.
